### PR TITLE
Feature/show history

### LINKED ISSUE #2: Show History
It'd be cool to see a list of all the searches. Maybe even make them clickable. History does not persist beyond browser session.

screenshot

### PR DESCRIPTION
**Pull request by icegotcha**
Closes #2 

### Description

I develop the feature as follow:
- Implement a function to store search terms. Use `sessionStorage`.
- Add the history section to the site header.
- Add a responsive modal to show a list of all used search terms to prevent show too many search terms on the site header. (I use [react-modal](https://www.npmjs.com/package/react-modal))

Besides, I also made a minor change:
- Add `browser: true` to `env` in eslint.js to prevent show errors (for example: "window is not defined")

### Screenshot

https://imgur.com/a/205VQT9

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ module.exports = {
   },
   env: {
     node: true,
+    browser: true,
```
Comment (gregrickaby) [inline .eslintrc.js:11]: 👏🏻 👏🏻 👏🏻

File: styles/index.css [hunk 2]
```diff
@@ -173,3 +184,68 @@ body.dark-mode {
 .theme-toggle-icon {
   @apply ml-1;
 }
+
+/* Modal
+============================== */
+
+.overlay {
```
Comment (gregrickaby) [inline styles/index.css:191]: @icegotcha Please use Tailwind utility classes whenever possible. Custom CSS should be used as a last resort. Thanks.

File: styles/index.css [hunk 3]
```diff
@@ -173,3 +184,68 @@ body.dark-mode {
...
+}
+
+.modal {
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  border: 1px solid #ccc;
+  background: #fff;
+  /* overflow: hidden; */
```
Comment (gregrickaby) [inline styles/index.css:207]: @icegotcha Please remove commented out code.

File: pages/index.js [hunk 4]
```diff
@@ -190,6 +301,9 @@ export default function Homepage() {
         <ThemeToggle />
         <BackToTop />
       </main>
+      {renderHistoryModal(searchHistory, showHistoryModal, () => {
```
Comment (gregrickaby) [inline pages/index.js:304]: @icegotcha I think there are too many things in the nav...and would prefer if history was hidden _until_ I click the history button.

**Review by gregrickaby — CHANGES_REQUESTED** [review]

File: styles/index.css (hunk 2 above)
Comment (icegotcha) [reply styles/index.css:187]: Done.

File: styles/index.css (hunk 3 above)
Comment (icegotcha) [reply styles/index.css]: Done.

File: pages/index.js (hunk 4 above)
Comment (icegotcha) [reply pages/index.js:263]: Done. I removed the third line in the nav and place the button next to example search terms.

**Review by gregrickaby — APPROVED** [review]

**Pull request merged**